std/gates/treegates.act

```diff
@@ -330,6 +330,37 @@ defproc sigbuf_gen (bool in; bool out[N])
 export defproc sigbuf <: sigbuf_gen<0> () { }
 export defproc sigbufa <: sigbuf_gen<1> () { }
 
+defcell xor2 (bool? in[2]; bool! out)
+{
+   bool _in0, _in1;
+   prs {
+     in[0] => _in0-
+     in[1] => _in1-
+
+    [keeper=2] ~in[1] & ~_in0 | ~_in1 & ~in[0] -> out+
+     _in1 & _in0 | in[1] & in[0] -> out-
+   }
+   sizing { _in0{-1}; _in1{-1}; out{-1} }
+}
+
+/*
+ * Build an XOR-gate tree with optional inversion of final output
+ */
+template <pint N; pbool invert> 
```

Comment:
It looks like `invert` is an unused parameter?

Reply:
Oops..
Fixed it